Rewrite `Beatmap editor/Distance snap` and `Mapping techniques/Time-distance equality` (#9864)
ppy/osu-wiki: wiki/Beatmapping/Mapping_techniques/Time-distance_equality/en.md

---
tags:
  - distance snap
  - DS
---

# Time-distance equality

::: Infobox
![](/wiki/shared/editor/editor-draw-distsnap.png "The distance snap tool icon")
:::

**Time-distance equality** is an [osu!](/wiki/Game_mode/osu!) [mapping technique](/wiki/Beatmapping/Mapping_techniques) where spacing between [hit objects](/wiki/Gameplay/Hit_object) on the playfield is proportional to the time between them. In the beatmap editor, it may be enforced by using the [distance snap](/wiki/Client/Beatmap_editor/Distance_snap) tool, which maintains the proportion according to the distance spacing multiplier.

## Usage

> The equality principle is built around the idea that visual density of hit objects should correspond to the song's structure: intensive rhythms mean that objects should be located near each other, and vice versa. Applying time-distance equality makes the beatmap more readable and often intuitive, since it provides consistent visual cues as to when hit objects should be clicked.

Reviewer: ```suggestion
The equality principle is built around the idea that visual density of hit objects should correspond to the song's structure: faster rhythm means that objects should be located near each other, and vice versa. Applying time-distance equality makes the beatmap more readable and often intuitive, since it provides consistent visual cues as to when hit objects should be clicked.
```

Reviewer 2: shouldn't this have been "a faster rhythm" or "faster rhythms"?

While on lower difficulties it is highly recommended,[^rc-note] harder difficulties strictly following said principle gradually become less exciting for experienced players due to predictability. To counter that, mappers often create [jumps](/wiki/Beatmapping/Mapping_techniques/Jumps) of varying extremity to add extra movement opportunities and bring variety.

### Benefits

> Although the time-distance equality is rarely used in the exact sense throughout the whole difficulty, it may often prove useful:

Reviewer: ```suggestion
Although time-distance equality is rarely used in the exact sense throughout the whole difficulty, it may often prove useful:
```

- Generally, the proportion improves readability of adjacent [beats](/wiki/Music_theory/Beat) with different [snapping](/wiki/Client/Beatmap_editor/Beat_snap), which is appreciated in sections with complex, alternating, or unintuitive [rhythm](/wiki/Music_theory/Rhythm). 

> - Using the equality produces [patterns](/wiki/Beatmap/Pattern) that are visually balanced, for example, regular polygons, or regular [stream](/wiki/Beatmap/Pattern/osu!/Stream) sections.

Reviewer 3: beyond 5 stars time-distance equality doesn't really have much to do with streams

bursts and streams in 4 and 5 stars usually have consistent super low spacing

Author: evenly-spaced streams are a good example of the time-distance equality IMO, and that is what I wanted to mention here

> - Using the equality produces [patterns](/wiki/Beatmap/Pattern) that are visually balanced, for example, regular polygons, or regular [stream](/wiki/Beatmap/Pattern/osu!/Stream) sections;

Reviewer: ```suggestion
- Using the equality produces [patterns](/wiki/Beatmap/Pattern) that are visually balanced, for example, regular polygons, or regular [stream](/wiki/Beatmap/Pattern/osu!/Stream) sections.
```
